21622-Enhance-FileReference-API-to-allow-to-choose-encoding

guillep opened the pull request:
- Add readStreamEncoded: and writeStreamEncoded: on AbstractFileReference
 - Removed need for encoded read and write streams in FileSystemHandle (encoding is done in the file reference).
 - FileSystemHandle should only provide binaryReadStream and binaryWriteStream

 - Update tests for all these changes.

guillep committed:
Fix #21622: Enhance FileReference API to allow to choose encoding
 - Add readStreamEncoded: and writeStreamEncoded: on AbstractFileReference
 - Removed need for encoded read and write streams in FileSystemHandle (encoding is done in the file reference).
 - FileSystemHandle should only provide binaryReadStream and binaryWriteStream

 - Update tests for all these changes.

jecisc commented:
This is cool :) We will be able to simplify Moose's code with this kind of changes.

dionisiydk commented on src/FileSystem-Tests-Core/FileSystemTest.class.st:
```diff
@@ -447,7 +447,7 @@ FileSystemTest >> testFile [
 	self markForCleanup: path.
 	
-	(filesystem open: path writable: true) writeStream close.
+	(filesystem workingDirectory resolve: path) writeStream close.
```
It would be nice to avoid #workingDirectory usage for tests because actual folder can not be writable.
For example if I will run image using Finder on Mac these tests will fail because directory will be root (/).

We should use FileLocator temp if possible

guillep replied:
Yeh, but I just rewrote the tests to a version equivalent to what they did before.
I did not refactor them...

Probably we should do that in another issue.

dionisiydk replied:
Agree. I opened https://pharo.fogbugz.com/f/cases/21625/We-should-avoid-workingDirectory-to-be-used-for-temp-test-files

svenvc commented:
Good

guillep committed:
Fix tests
 - Move buffering from handle to reference => this adds buffering support to memory file system
 - Enhance memory file system	
 - Enhance Source files to be able to use a memory file system and be thus be buffered too
 - System settings should close its write stream
 - GT event recorder tests should use new binary stream API

guillep committed:
close file streams only if open
